src/common/state/redis.cc

```diff
@@ -633,31 +751,33 @@ void redis_object_table_request_notifications(
   int num_object_ids = request_data->num_object_ids;
   ObjectID *object_ids = request_data->object_ids;
 
-  /* Create the arguments for the Redis command. */
-  int num_args = 1 + 1 + num_object_ids;
-  const char **argv = (const char **) malloc(sizeof(char *) * num_args);
-  size_t *argvlen = (size_t *) malloc(sizeof(size_t) * num_args);
-  /* Set the command name argument. */
-  argv[0] = "RAY.OBJECT_TABLE_REQUEST_NOTIFICATIONS";
-  argvlen[0] = strlen(argv[0]);
-  /* Set the client ID argument. */
-  argv[1] = (char *) db->client.id;
-  argvlen[1] = sizeof(db->client.id);
-  /* Set the object ID arguments. */
   for (int i = 0; i < num_object_ids; ++i) {
-    argv[2 + i] = (char *) object_ids[i].id;
-    argvlen[2 + i] = sizeof(object_ids[i].id);
-  }
-
-  int status = redisAsyncCommandArgv(
-      db->context, redis_object_table_request_notifications_callback,
-      (void *) callback_data->timer_id, num_args, argv, argvlen);
-  free(argv);
-  free(argvlen);
-
-  if ((status == REDIS_ERR) || db->context->err) {
-    LOG_REDIS_DEBUG(db->context,
-                    "error in redis_object_table_subscribe_to_notifications");
+    redisAsyncContext *context = get_redis_context(db, object_ids[i]);
```

Comment:
Oh, I realize now that you're doing this because the different requests need to go to different redis shards. However, in the case where we are requesting 1 million objects, this may be too expensive..